src/plugins/navigation_embeddable/public/components/navigation_embeddable_panel_editor.tsx

```diff
@@ -0,0 +1,197 @@
+
+export const NavigationEmbeddablePanelEditor = ({
+  onSave,
+  onClose,
+  initialInput,
+  parentDashboard,
+}: {
+  onClose: () => void;
+  initialInput: Partial<NavigationEmbeddableInput>;
+  onSave: (input: Partial<NavigationEmbeddableInput>) => void;
+  parentDashboard?: DashboardContainer;
+}) => {
+  const [showLinkEditorFlyout, setShowLinkEditorFlyout] = useState(false);
+  const [links, setLinks] = useState(initialInput.links);
+
+  /**
+   * TODO: There is probably a more efficient way of storing the dashboard information "temporarily" for any new
+   * panels and only fetching the dashboard saved objects when first loading this flyout.
+   *
+   * Will need to think this through and fix as part of the editing process - not worth holding this PR, since it's
+   * blocking so much other work :)
+   */
+  const { value: linkList } = useAsync(async () => {
+    if (!links || isEmpty(links)) return [];
+
+    const newLinks: Array<{ id: string; icon: IconType; label: string }> = await Promise.all(
+      Object.keys(links).map(async (panelId) => {
+        let label = links[panelId].label;
+        let icon = NavigationLinkInfo[EXTERNAL_LINK_TYPE].icon;
+
+        if (links[panelId].type === DASHBOARD_LINK_TYPE) {
+          icon = NavigationLinkInfo[DASHBOARD_LINK_TYPE].icon;
+          if (!label) {
+            const dashboard = await memoizedFetchDashboard(links[panelId].destination);
+            label = dashboard.attributes.title;
+          }
+        }
+
+        return { id: panelId, label: label || links[panelId].destination, icon };
+      })
+    );
+    return newLinks;
+  }, [links]);
+
+  return (
+    <>
+      <EuiFlyoutHeader hasBorder>
+        <EuiTitle size="m">
+          <h2>{NavEmbeddableStrings.editor.panelEditor.getCreateFlyoutTitle()}</h2>
+        </EuiTitle>
+      </EuiFlyoutHeader>
+      <EuiFlyoutBody>
+        <EuiForm fullWidth>
+          <EuiFormRow>
+            <>
+              {!links || Object.keys(links).length === 0 ? (
+                <EuiPanel hasBorder={true}>
+                  <EuiFlexGroup justifyContent="spaceAround">
+                    <EuiFlexItem grow={false}>
+                      <EuiText size="s">
+                        {NavEmbeddableStrings.editor.panelEditor.getEmptyLinksMessage()}
+                      </EuiText>
+                    </EuiFlexItem>
+                  </EuiFlexGroup>
+                  <EuiSpacer size="s" />
+                  <EuiFlexGroup justifyContent="spaceAround">
+                    <EuiFlexItem grow={false}>
+                      <EuiButton
+                        onClick={() => setShowLinkEditorFlyout(true)}
+                        iconType="plusInCircle"
+                      >
+                        {NavEmbeddableStrings.editor.getAddButtonLabel()}
+                      </EuiButton>
+                    </EuiFlexItem>
+                  </EuiFlexGroup>
+                </EuiPanel>
+              ) : (
```

Comment:
Doesn't have to block this PR, but I wonder if we should (can?) add a nice [empty prompt](https://elastic.github.io/eui/#/display/empty-prompt) to the flyout? It would be nice to have a title prompt like "Create links" and a description explaining the value of the links panel, e.g. "Links can pass filters and time range to other dashboards. Or create external links to other websites." <-- probably needs some better copy 😄

Reply:
I 100% agree - @andreadelrio @gchaps any ideas on how to add some visual interest to the flyout and/or improve the copy to give it more impact when there are no links? :eyes: For reference, this is what the flyout currently looks like when first creating a links panel:

![image](https://github.com/elastic/kibana/assets/8698078/a213deb0-8abc-401c-a348-431ba011cf5d)

I definitely think there's work we could do to make it a little more ⭐ fancy ⭐ even if it's not included as part of this PR :)

Reply:
@Heenawter we can definitely improve the text.  Just watched the video.  Is this a panel in a dashboard that can link to other dashboards and websites?

Reply:
@gchaps Yes, it is! :+1: It is still in its early stages, but we are essentially creating a navigation system so that users don't have to use markdown links (which can very easily become out of date, especially when using hard links to specific dashboards) anymore. I can link you to the project docs on Slack if you want more information - sorry for the sudden no-context mention, haha! 😅

Reply:
I agree that we could make this more visually appealing. Let me think about what we can do here. I don't think we have any illustration available right now that would fit the bill so there might not be a quick solution to this.

Reply:
@nickpeihl @Heenawter @gchaps sending a couple of ideas for the empty state graphic.

<img width="659" alt="image" src="https://github.com/elastic/kibana/assets/4016496/1990eb1f-00d9-41fe-928c-d39fc31c302f">
<img width="660" alt="image" src="https://github.com/elastic/kibana/assets/4016496/7b168e67-d84b-4d91-a3c2-6c9cf54c6920">

Reply:
Wow! The first one with the big link and the fly looks incredible!

Reply:
I'm partial to the first graphic.

Reply:
Oooh, those are fantastic Andrea :fire: I especially like the first one!! :) I've opened a related [issue](https://github.com/elastic/kibana/issues/161552) so that we can move discussion of the empty state there (both for copy suggestions and the graphic) - I think it's worth getting multiple opinions, and this thread is already getting pretty large hehe